skipper-ingress: disable automated healthcheck routes

ponimas opened the pull request:
Skipper provides two flags to control kubernetes healthcheck routes:
`-kubernetes-healthcheck` (true by default) that enables automatic healthcheck routes
and `-reverse-source-predicate` to select source predicate type.

Currently when routesrv is enabled (default state) it adds
automatic healthcheck routes and appends default filters to 
all routes including healthcheck routes.

We want  to avoid adding default filters to healthcheck routes. The explicit inlined checks were created in [this PR](https://github.com/zalando-incubator/kubernetes-on-aws/pull/9206). 
This change disables automatic healthcheck routes provided by routesrv.

Additionally this PR add an iam anotation to these healhcheck routes to opt-out them from the authentication.

- [x] ~This PR should be merged only after [this one](https://github.com/zalando-incubator/kubernetes-on-aws/pull/9206) is deployed.~

ponimas committed:
skipper-ingress: disable automated healthcheck routes at routesrv

Skipper provides two flags to control kubernetes healthcheck routes:
`-kubernetes-healthcheck` (true by default) that enables automatic healthcheck routes
and `-reverse-source-predicate` to select source predicate type.

Currently when routesrv is enabled (default state) it adds
automatic healthcheck routes and appends default filters to
all routes including healthcheck routes.

We want  to avoid adding default filters to healthcheck routes. The explicit inlined checks were created in #9206
This change disables automatic healthcheck routes provided by routesrv.

This reverts commit d6efa246a69bffdbe85f47de772f86ad545a7e11.

Signed-off-by: Aleksandr Ponimaskin <aleksandr.ponimaskin@zalando.de>

AlexanderYastrebov commented on cluster/manifests/skipper/deployment.yaml:
```diff
@@ -578,7 +579,6 @@ spec:
           - '-kubernetes-annotation-filters-append={{ .Cluster.ConfigItems.skipper_kubernetes_annotation_filters_append }}'
           - '-kubernetes-east-west-range-annotation-predicates={{ .Cluster.ConfigItems.skipper_kubernetes_east_west_range_annotation_predicates }}'
           - '-kubernetes-east-west-range-annotation-filters-append={{ .Cluster.ConfigItems.skipper_kubernetes_east_west_range_annotation_filters_append }}'
-          - "-reverse-source-predicate"
```
We'd also need to drop `-whitelisted-healthcheck-cidr`, see https://github.com/zalando-incubator/kubernetes-on-aws/pull/9308

ponimas committed:
refactor: Remove IPv6 healthcheck CIDR whitelist config

Remove the conditional configuration for IPv6 healthcheck CIDR
whitelisting for routeserv in the skipper deployment manifest. This
option is no longer necessary as the healthchecks are moved to the
skipper-ingress level.

Signed-off-by: Aleksandr Ponimaskin <aleksandr.ponimaskin@zalando.de>

AlexanderYastrebov commented:
:+1:

ponimas commented:
👍

AlexanderYastrebov commented:
> We want to avoid adding default filters to healthcheck routes

With routesrv disabled Skipper will add default filters to healthcheck routes. 
Lets hold on and think about how to make this safe.

ponimas committed:
Add IAM annotation to healthcheck routes

We need to exclude healthcheck routes from the authentication even
cases when the routeserv is disabled to keep the backwards compartibility.

AlexanderYastrebov commented on cluster/manifests/skipper/deployment.yaml:
```diff
@@ -326,11 +326,17 @@ spec:
           - |
             kube__healthz_down: Path("/kube-system/healthz") && Shutdown()
               && SourceFromLast("{{ .cluster_internal_cidrs | join `","` }}", "10.0.0.0/8", "192.168.0.0/16", "172.16.0.0/12", "127.0.0.1/8", "fd00::/8", "::1/128")
+{{ if ne .Cluster.ConfigItems.skipper_routesrv_enabled "exec" }}
```
Lets maybe have this unconditionally for simplicity?

ponimas committed:
refactor: Remove conditional annotations from healthz routes

Signed-off-by: Aleksandr Ponimaskin <aleksandr.ponimaskin@zalando.de>

AlexanderYastrebov commented:
:+1:

ponimas commented:
👍
